src/extensions/extensions-store.ts

```diff
@@ -0,0 +1,49 @@
+import type { LensExtensionId } from "./lens-extension";
+import { BaseStore } from "../common/base-store"
+import { action, observable, toJS } from "mobx";
+
+export interface LensExtensionsStoreModel {
+  extensions: Record<LensExtensionId, LensExtensionState>;
+}
+
+export interface LensExtensionState {
+  enabled?: boolean;
+}
+
+export class ExtensionsStore extends BaseStore<LensExtensionsStoreModel> {
+  constructor() {
+    super({
+      configName: "lens-extensions"
+    });
+  }
+
+  @observable extensions = observable.map<LensExtensionId, LensExtensionState>();
+
+  @action
+  setEnabled(extId: LensExtensionId, enabled: boolean) {
+    const state = this.extensions.get(extId);
+    this.extensions.set(extId, {
+      ...(state || {}),
+      enabled: enabled,
+    })
```

Comment:
We are talking about observable objects here, I don't believe that all this code is necessary. 
```suggestion
    this.extensions.get(extId).enabled = enabled
```

Reply:
State is empty for all extensions by default (they are enabled)

Reply:
Ah I see